pkg/abaputils/abaputils_test.go

```diff
@@ -0,0 +1,69 @@
+package abaputils
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadServiceKeyAbapEnvironment(t *testing.T) {
+	t.Run("CF ReadServiceKeyAbapEnvironment", func(t *testing.T) {
+
+		//given
+		options := AbapEnvironmentOptions{
+			Username:          "testUser",
+			Password:          "testPassword",
+			CfAPIEndpoint:     "https://api.endpoint.com",
+			CfSpace:           "testSpace",
+			CfOrg:             "testOrg",
+			CfServiceInstance: "testInstance",
+			CfServiceKeyName:  "testKey",
+		}
+
+		//when
+		var abapKey AbapServiceKey
+		abapKey, _ = ReadServiceKeyAbapEnvironment(options, &command.Command{}, true)
+
+		//then
+		assert.Equal(t, "", abapKey.Abap.Password)
+		assert.Equal(t, "", abapKey.Abap.Username)
+		assert.Equal(t, "", abapKey.Abap.CommunicationArrangementID)
+		assert.Equal(t, "", abapKey.Abap.CommunicationScenarioID)
+		assert.Equal(t, "", abapKey.Abap.CommunicationSystemID)
+		assert.Equal(t, "", abapKey.Binding.Env)
+		assert.Equal(t, "", abapKey.Binding.Type)
+		assert.Equal(t, "", abapKey.Binding.ID)
+		assert.Equal(t, "", abapKey.Binding.Version)
+		assert.Equal(t, "", abapKey.SystemID)
+		assert.Equal(t, "", abapKey.URL)
+		//assert.Error(t, err)
```

Comment:
This was commented out for testing purposes. But would the error be filled or empty?